src/internal/document_api/mod.rs

```diff
@@ -435,53 +500,169 @@ pub fn encrypt_document<
+    Item = (
+        //TODO: new ticket for returning vec1 here or error
+        Vec<WithKey<UserOrGroup>>,
+        Vec<DocAccessEditErr>,
+    ),
+    Error = IronOxideErr,
+> + 'a {
     internal::user_api::get_user_keys(auth, user_grants)
-        .join4(
+        .join3(
+            // TICKET: only make these calls if the vecs are non-empty
             internal::group_api::get_group_keys(auth, group_grants),
             policy_grant.map(|p| requests::policy_get::policy_get_request(auth, p)),
-            aes::encrypt_future(rng, &plaintext.to_vec(), *doc_sym_key.bytes()),
         )
-        .and_then(move |(users, groups, maybe_policy_res, encrypted_doc)| {
+        .map(move |(users, groups, maybe_policy_res)| {
             let (group_errs, groups_with_key) = process_groups(groups);
             let (user_errs, users_with_key) = process_users(users);
-            let explicit_grants = if grant_to_author {
-                let self_grant = WithKey::new(
-                    UserOrGroup::User {
-                        id: auth.account_id.clone(),
-                    },
-                    user_master_pub_key.clone(),
-                );
-                [&[self_grant], &users_with_key[..], &groups_with_key[..]].concat()
-            } else {
-                [&users_with_key[..], &groups_with_key[..]].concat()
-            };
+            let explicit_grants = [users_with_key, groups_with_key].concat();
             let (policy_errs, applied_policy_grants) = match maybe_policy_res {
                 None => (vec![], vec![]),
                 Some(res) => process_policy(&res),
             };
+            let maybe_self_grant = {
+                if let Some(user_master_pub_key) = maybe_user_master_pub_key {
+                    vec![WithKey::new(
+                        UserOrGroup::User {
+                            id: auth.account_id.clone(),
+                        },
+                        user_master_pub_key.clone(),
+                    )]
+                } else {
+                    vec![]
+                }
+            };
 
-            // squish all accumulated errors into one list
-            let other_errs = vec![group_errs, user_errs, policy_errs]
-                .into_iter()
-                .concat();
-
-            encrypt_document_core(
-                auth,
-                recrypt,
-                dek,
-                encrypted_doc,
-                doc_id,
-                document_name,
-                [explicit_grants, applied_policy_grants].concat(),
-                other_errs,
+            (
+                { [maybe_self_grant, explicit_grants, applied_policy_grants].concat() },
+                [group_errs, user_errs, policy_errs].concat(),
             )
         })
 }
 
+/// Encrypts a document but does not create the document in the IronCore system.
+/// The resultant DocumentDetachedEncryptResult both the EncryptedDeks and the AesEncryptedValue for the caller to deal with.
```

Comment:
Nitpick - this sentence is missing a verb, and it's making it slightly less clear to me. I assume it means "The resultant DocumentDetachedEncryptResult **contains** both the..."?